### PR TITLE
Adding Pattern Matching app

### PR DESCRIPTION
**Pull request by IcToxi**

### CODE REVIEW

**Comment by IcToxi** [comment]
image

**Comment by IcToxi** [comment]
image

**Comment by IcToxi** [comment]
There's something wrong with the table's response to hover.

**Comment by AnnMarieW** [comment]
The datatable can be tricky to style.  For the hover, I usually just turn it off.  The css below works when it's in the assets folder, but I haven't yet figured out how to make it work in `css` parameter of the DataTable.  (There may be other css that's overriding it)

```

 .dash-table-container .dash-spreadsheet-container .dash-spreadsheet-inner table {
             --hover: transparent !important;
 }
```

Another option is to use the [dbc css stylesheet](https://hellodash.pythonanywhere.com/about_dbc_css) I developed - but this might be beyond the scope of this project as well.

**Comment by IcToxi** [comment]
I don't know why, when I set it to this below, it worked.
```python
{
    "selector": "table",
     "rule": "--hover: tomato;",
},
```
There is still one problem left unresolved, that is, when I click on a few cells, the color of the out-of-focus cells is wrong. I'm not sure if that's a bug.

And the chart's title font size seems to only accept numeric values, and I haven't found a way to set its relative size.

**Comment by IcToxi** [comment]
> There is still one problem left unresolved, that is, when I click on a few cells, the color of the out-of-focus cells is wrong. I'm not sure if that's a bug.

I fixed this with the selector `.dash-cell`.

File: pages/examples/questionnaire.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+import requests
+
+app = Dash(__name__)
+
+questionnaire = requests.get(
+    "https://raw.githubusercontent.com/IcToxi/datasets/main/questionnaire.json"
+).json()
+
+
+def generate(k, v):
```
Comment (IcToxi) [inline pages/examples/questionnaire.py:12]: `def generate(k: str, v: dict) -> ??:`

**Comment by Coding-with-Adam** [comment]
HI @IcToxi 
I incorporated the parallel coordinates app into the gallery. In regard to the other apps, can you modify the `map-with-filter.py` file so it is under 100 lines of code. We also need to declare IDs in the callback because the gallery cannot read autogenerated IDs. They have to be `name-of-file-x-`

The table-with-barpolar.py app is also very useful but here too can we reduce the code to under 100 lines and have the IDs be `name-of-file-x-`?

Regarding the pattern matching app (questionnaire.py), because of `match` and `case` it requires Python 3.10. @AnnMarieW if I add this cool app, and people don't run the gallery with Python 3.10, will it break the whole gallery or just this particular app won't work for them?

**Comment by IcToxi** [comment]
Hi @Coding-with-Adam,

I think the original intent of `filename-x-` is to avoid duplicate IDs in multipage app,  while random IDs do not have this problem. I'm not sure what you mean by not being able to read them. Actually, I think that's an issue with `_run_code`, there is no necessary to execute `compile`. Furthermore, I noticed that the gallery app also uses random IDs in `dcc.Clipboard`, but that seems to be py376 usage. How long ago did you write this stuff?

The code that occupies space in that `table-with-barpolar` mainly comes from the part that integrates with the theme. I tried this because I saw a lot of people doing dark mode. If I continue to simplify it, then I will do it in the direction of stripping the theme. As for `map-with-filter`, if it continues to be simplified, I consider cutting off the `with-filter` part, but then it will be no different from the official documentation. Are you sure I should do this?

This match-case assignment issue is an adaptation difficulty, there are two ways to solve it, change my code to if-else assignment, or make this gallery project compatible. Obviously, this will not be the first time this issue has been encountered. So, as a framework-type app, compatibility should be required.

And I think as a community application, it should be more aggressive. Those with a learning mindset should learn from here what they are really interested in, rather than cookie-cutter documentation. And our roles in this project are different, you and @AnnMarieW have to do more to enhance the adaptability and ease of use of this project. Instead of making more demands on example contributors, so that you can attract more people to contribute to this.

**Comment by IcToxi** [comment]
I think you guys should consider completely redesigning this project, obviously the current structure can't support all the items we put on the list. It would be a good choice to use Flask to implement the multipage structure and allow the creation of new Dash instances. Otherwise, features including themes, plugins, CSS, etc. will not have the chances to show.

**Comment by Coding-with-Adam** [comment]
> I think you guys should consider completely redesigning this project, obviously the current structure can't support all the items we put on the list. It would be a good choice to use Flask to implement the multipage structure and allow the creation of new Dash instances. Otherwise, features including themes, plugins, CSS, etc. will not have the chances to show.

Interesting suggestion @IcToxi . I want to make sure that we finalize and feel confident and comfortable with version 1.0 of the current structure we put in place. Then, for version 2.0 we can look into expanding the examples and teaching opportunities.

**Pull request closed without merging**